### PR TITLE
feat: add new proto definitions for tokenless, remove the need for SetGRPCConnection

### PR DESCRIPTION
**Pull request by cameronmeissner**

### CODE REVIEW

File: pkg/protos/bootstrap.proto [hunk 1]
```diff
@@ -7,34 +7,59 @@ option go_package = "github.com/Azure/aks-tls-bootstrap-client/protos";
...
+
+// DEPRECATED: A token request has to match a valid generated nonce and auth data.
 message TokenRequest {
     string ResourceId = 1;
     string Nonce = 2;
     string AttestedData = 3;
 }
 
-// The response token is used by kubelet to bootstrap itself.
+// DEPRECATED: The response token is used by kubelet to bootstrap itself.
```
Comment (cameronmeissner) [inline pkg/protos/bootstrap.proto:62]: will follow-up with removal of these in subsequent PRs

File: pkg/client/service.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	opts serviceClientFactoryOpts) (secureTLSBootstrapService.SecureTLSBootstrapServiceClient, grpcClientConn, error) {
+	logger.Debug("decoding cluster CA data...")
+	pemCAs, err := base64.StdEncoding.DecodeString(opts.execCredential.Spec.Cluster.CertificateAuthorityData)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode base64 cluster certificates: %w", err)
+	}
+	logger.Info("decoded cluster CA data")
+
+	logger.Debug("generating TLS config for GRPC client connection...")
+	tlsConfig, err := getTLSConfig(pemCAs, opts.nextProto, opts.execCredential.Spec.Cluster.InsecureSkipTLSVerify)
```
Comment (bcho) [inline pkg/client/service.go:43]: nit: log the tls config before calling for debugging purposes?

File: pkg/client/log.go [hunk 3]
```diff
@@ -31,7 +34,7 @@ func GetLogger(format string, debug bool) (*zap.Logger, error) {
 }
 
 func FlushBufferOnExit(logger *zap.Logger) {
-	if err := logger.Sync(); err != nil {
+	if err := logger.Sync(); err != nil && !errors.Is(err, syscall.ENOTTY) {
```
Comment (bcho) [inline pkg/client/log.go:37]: what does this `ENOTTY` mean?

Comment (cameronmeissner) [reply pkg/client/log.go:37]: we need this for when stdout/stderr is redirected to the console, such as during unit tests

this is a known issue with Zap unfortunately: https://github.com/uber-go/zap/issues/880#issuecomment-1181854418

Comment (bcho) [reply pkg/client/log.go:37]: add a note for reference?

Comment (cameronmeissner) [reply pkg/client/log.go:37]: will add a clarifying comment

Comment (bcho) [reply pkg/client/log.go:37]: and maybe refine the code a bit to exit early

```
syncErr := logger.Sync()
if syncErr == nil {
  return
}

switch {
  case errors.Is(err, syscall.ENOTTY):
    // known issue
    return
  default:
    // log
}
```

**Review by bcho — APPROVED** [review]
LGTM

**Review by AlisonB319 — APPROVED** [review]
lgtm

**Comment by cameronmeissner** [comment]
coveralls is upset due to added handling in zap logger flush

**Pull request merged**